proxy.go

```diff
@@ -69,22 +76,17 @@ func (ph ProxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 func (ph ProxyHandler) handleConnect(w http.ResponseWriter, req *http.Request) {
 	// Establish a connection to the server, or an upstream proxy.
-	u, err := ph.transport.Proxy(req)
 	id := req.Context().Value(contextKeyID)
-	if err != nil {
-		log.Printf("[%d] Error finding proxy for %v: %v", id, req.Host, err)
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
 	var server net.Conn
-	if u == nil {
+	var err error
+	if proxy, _ := ph.transport.Proxy(req); proxy == nil {
```

Comment:
you're ignoring the error from `ph.transport.Proxy()` - is that to fallback to a direct connection with `net.Dial()` if you get an error with the proxy? Should that at least be logged?

Reply:
Yes, that was the original intention. Agree it should be logged, I've just added this.